### ISSUE TITLE
关于使用pgnet用于识别车牌的一些问题

### ISSUE THREAD

**Issue by 8125345** [PaddlePaddle/PaddleOCR#4312]
在用pgnet识别车牌遇到一些问题，
我的数据格式：
![image](https://user-images.githubusercontent.com/53589072/137061259-ac690fb4-f91e-44a6-8b81-170d23f0121d.png)
我的配置文件：（只改了几个文件路径）
Global:
  use_gpu: True
  epoch_num: 600
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/pgnet_r50_vd_lincenceplate/
  save_epoch_step: 10
  # evaluation is run every 1000 iterationss after the 0th iteration
  eval_batch_step: [ 0, 1000 ]
  cal_metric_during_train: False
  pretrained_model:
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img:
  valid_set: totaltext # two mode: totaltext valid curved words, partvgg valid non-curved words
  save_res_path: ./output/pgnet_r50_vd_lincenceplate/predicts_pgnet.txt
  character_dict_path: ppocr/utils/lincenceplate.txt
  character_type: ch
  max_text_length: 50 # the max length in seq
  max_text_nums: 30 # the max seq nums in a pic
  tcl_len: 64
Architecture:
  model_type: e2e
  algorithm: PGNet
  Transform:
  Backbone:
    name: ResNet
    layers: 50
  Neck:
    name: PGFPN
  Head:
    name: PGHead

Loss:
  name: PGLoss
  tcl_bs: 64
  max_text_length: 50 # the same as Global: max_text_length
  max_text_nums: 30 # the same as Global：max_text_nums
  pad_num: 36 # the length of dict for pad

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: PGPostProcess
  score_thresh: 0.5
  mode: fast   # fast or slow two ways

Metric:
  name: E2EMetric
  mode: A   # two ways for eval, A: label from txt,  B: label from gt_mat
  gt_mat_dir:  ./train_data/total_text/gt  # the dir of gt_mat
  character_dict_path: ppocr/utils/lincenceplate.txt
  main_indicator: f_score_e2e

Train:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/licenceplate/train
    label_file_list: [./train_data/licenceplate/train/train.txt]
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - E2ELabelEncodeTrain:
      - PGProcessTrain:
          batch_size: 14  # same as loader: batch_size_per_card
          min_crop_size: 24
          min_text_size: 4
          max_text_size: 512
      - KeepKeys:
          keep_keys: [ 'images', 'tcl_maps', 'tcl_label_maps', 'border_maps','direction_maps', 'training_masks', 'label_list', 'pos_list', 'pos_mask' ] # dataloader will return list in this order
  loader:
    shuffle: True
    drop_last: True
    batch_size_per_card: 14
    num_workers: 16

Eval:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/licenceplate/test
    label_file_list: [./train_data/licenceplate/test/test.txt]
    transforms:
      - DecodeImage: # load image
          img_mode: RGB
          channel_first: False
      - E2ELabelEncodeTest:
      - E2EResizeForTest:
          max_side_len: 768
      - NormalizeImage:
          scale: 1./255.
          mean: [ 0.485, 0.456, 0.406 ]
          std: [ 0.229, 0.224, 0.225 ]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: [ 'image', 'shape', 'polys', 'texts', 'ignore_tags', 'img_id']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 2
我的字典是将数字和大写字母，省份简称写进了txt里，
我的训练过程和训练过程的评估结果：
![image](https://user-images.githubusercontent.com/53589072/137061562-ba1156f8-984d-412b-824c-4a5d2724783c.png)

然后训练了600个epoch，预测结果如下：（都是错的）
![20210824050558川AQ2517-2](https://user-images.githubusercontent.com/53589072/137061288-6fbf6754-352b-432b-acb8-3af54060fd58.jpg)
![20210824051608青H60841-2](https://user-images.githubusercontent.com/53589072/137061299-8cd12e4a-6416-464c-a0d1-08542b6b0281.jpg)
![20210824051753川AAN168-2](https://user-images.githubusercontent.com/53589072/137061320-4874fa66-30b9-4c6f-8fbf-2d33dd01654b.jpg)
![20210824052954川T17191-2](https://user-images.githubusercontent.com/53589072/137061333-e9c717d0-ce8b-4344-9bd4-80d999be410c.jpg)
![20210824054354川AEL728-2](https://user-images.githubusercontent.com/53589072/137061348-71d39a51-f3bd-4a0b-bcf2-be82b492e965.jpg)
![20210824094903川AAY072](https://user-images.githubusercontent.com/53589072/137061363-a5bea7ad-3147-4f33-8f87-33fb3afa5ce3.jpg)
请问这种情况是不是应该修改什么配置呢？还是pgnet不适用于车牌识别？

**Comment by LDOUBLEV** [comment]
最近修复了一个pgnet图像通道的问题，可以按照修改后试下：https://github.com/PaddlePaddle/PaddleOCR/pull/4308/files

看上去是检测到了，是识别的问题，可以看下识别模块训练精度是否正常

**Comment by LDOUBLEV** [comment]
建议用我们的DB+CRNN方法，效果也不错

**Comment by yjmm10** [comment]
> 最近修复了一个pgnet图像通道的问题，可以按照修改后试下：https://github.com/PaddlePaddle/PaddleOCR/pull/4308/files
> 
> 看上去是检测到了，是识别的问题，可以看下识别模块训练精度是否正常

我在paddleocr 2.2版本尝试将RGB修改为BGR，结果模型无法运行出结果，RGB就能正常得到结果

**Comment by itruonghai** [comment]
@8125345 How can you train with 4 points only? I trained PGNet with 4 points and it raised an error of invalid poly.

**Comment by Ehteshamciitwah** [comment]
> DB+CRNN



> @8125345 How can you train with 4 points only? I trained PGNet with 4 points and it raised an error of invalid poly.

Did you find any solution. I also want to train on my dataset. But dont know how to format dataset with 14 points.

**Comment by windspirit95** [comment]
Maybe the training issue come from "valid_set": It should be partvgg instead of totaltext because the annotation format here is 4 points, not 14 point format of totaltext :)

**Comment by fangfangzk** [comment]
What is ’partvgg‘ ?
